### PR TITLE
Ensure docker can read artifacts in flexible deploys

### LINKED ISSUE #90: Ensure artifacts have the correct file permissions
After staging a flexible style project, we need to ensure the artifact is readable by everyone or the user launching it inside the docker wont be able to run it. Deploys fail to verify the vm is running and an 502 is presented when trying to hit the url.

### PR DESCRIPTION
**Pull request by loosebazooka**
Fixes #90

### CODE REVIEW

**Comment by loosebazooka** [comment]
@meltsufin

**Comment by meltsufin** [comment]
LGTM

File: src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java [hunk 1]
```diff
@@ -73,10 +77,20 @@ public void stageFlexible(StageFlexibleConfiguration config) throws AppEngineExc
...
-        Files.copy(config.getArtifact().toPath(),
-            config.getStagingDirectory().toPath()
-                .resolve(config.getArtifact().toPath().getFileName()),
-            REPLACE_EXISTING);
+        Path destination = config.getStagingDirectory().toPath()
+            .resolve(config.getArtifact().toPath().getFileName());
+        Files.copy(config.getArtifact().toPath(), destination, REPLACE_EXISTING);
+
+        // Update artifact permissions so docker can read it when deployed
+        if (!System.getProperty("os.name").contains("Windows")) {
```
Comment (patflynn) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java]: We're doing this differently in IJ: https://github.com/GoogleCloudPlatform/gcloud-intellij/blob/master/google-cloud-tools-plugin/src/com/google/cloud/tools/intellij/appengine/cloud/AppEngineDeployAction.java#L116

If this is the 'right' way we should probably use it there too.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java]: I don't think there's a right way, but there's definitely a much cleaner way. Will update and test.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java]: Oh I didn't see it used File and not Path. Let me look into this a little.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java]: So it appears Path is recommended going forward. I'm not sure how this is supposed to happen in Windows... added #91 for a set of Windows related checks we need to do.

Comment (loosebazooka) [inline src/main/java/com/google/cloud/tools/app/impl/cloudsdk/CloudSdkAppEngineFlexibleStaging.java]: Going to submit this for now, File looks way more convenient though... :\

**Pull request merged**